src/sources/internal_logs.rs

```diff
@@ -64,26 +65,26 @@ async fn run(
         .chain(tokio_stream::wrappers::BroadcastStream::new(
             subscription.receiver,
         ))
+        .filter_map(|log| future::ready(log.ok()))
         .take_until(shutdown);
 
     // Note: This loop, or anything called within it, MUST NOT generate
     // any logs that don't break the loop, as that could cause an
     // infinite loop since it receives all such logs.
-    while let Some(res) = rx.next().await {
-        match res {
-            Ok(mut log) => {
-                if let Ok(hostname) = &hostname {
-                    log.insert(host_key.clone(), hostname.to_owned());
-                }
-                log.insert(pid_key.clone(), pid);
-                log.try_insert(log_schema().source_type_key(), Bytes::from("internal_logs"));
-                log.try_insert(log_schema().timestamp_key(), Utc::now());
-                if let Err(error) = out.send(Event::from(log)).await {
-                    error!(message = "Error sending log.", %error);
-                    return Err(());
-                }
-            }
-            Err(BroadcastStreamRecvError::Lagged(_)) => (),
+    while let Some(mut log) = rx.next().await {
+        emit!(&EventsReceived {
```

Comment:
about the comment above (infinite loop), I tried to start vector with this source emitting this event, the events are not handled byte `internal_logs` and therefore do not trigger an infinite loop.

Reply:
Did you test this with logging set to trace? The `EventsReceived` event does emit a trace. In any case, this should have a comment as to why it is safe to do, and probably a comment on `EventsReceived` that it must not emit any logs due to this.